plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftUnloadSplitManager.java

```diff
@@ -0,0 +1,195 @@
+        }
+        ConnectorSplitSource fallbackSplitSource = new FixedSplitSource(new JdbcSplit(Optional.empty()));
+        if (!useUnload(session)) {
+            return fallbackSplitSource;
+        }
+        JdbcTableHandle jdbcTableHandle = dynamicFilteringEnabled(session) ? ((JdbcTableHandle) table).intersectedWithConstraint(dynamicFilter.getCurrentPredicate()) : (JdbcTableHandle) table;
+        List<JdbcColumnHandle> columns = jdbcTableHandle.getColumns()
+                .orElseGet(() -> jdbcClient.getColumns(
+                        session,
+                        jdbcTableHandle.getRequiredNamedRelation().getSchemaTableName(),
+                        jdbcTableHandle.getRequiredNamedRelation().getRemoteTableName()));
+
+        if (!isUnloadSupported(((JdbcTableHandle) table), columns)) {
+            log.debug("Unload query contains unsupported characters. Falling back to using JDBC");
+            return fallbackSplitSource;
+        }
+
+        Connection connection;
+        PreparedStatement statement;
+        try {
+            connection = jdbcClient.getConnection(session);
+            String redshiftSelectSql = getRedshiftSelectSql(session, connection, jdbcTableHandle, columns);
+            if (redshiftSelectSql.contains("\\b")) {
+                log.debug("Unload query contains unsupported characters. Falling back to using JDBC");
+                return fallbackSplitSource;
+            }
+            statement = buildUnloadSql(session, connection, columns, redshiftSelectSql);
+        }
+        catch (SQLException e) {
+            if (e instanceof RedshiftException) {
+                log.debug("Could not build unload query. Falling back to using JDBC");
+                return fallbackSplitSource;
+            }
+            throw new RuntimeException(e);
+        }
+        return new RedshiftUnloadSplitSource(executor, connection, statement, fallbackSplitSource);
+    }
+
+    private String getRedshiftSelectSql(ConnectorSession session, Connection connection, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        PreparedQuery preparedQuery = jdbcClient.prepareQuery(session, table, Optional.empty(), columns, ImmutableMap.of());
+        PreparedStatement openTelemetryPreparedStatement = queryBuilder.prepareStatement(jdbcClient, session, connection, preparedQuery, Optional.of(columns.size()));
+        RedshiftPreparedStatement redshiftPreparedStatement = openTelemetryPreparedStatement.unwrap(RedshiftPreparedStatement.class);
+        String selectQuerySql = redshiftPreparedStatement.toString();
+        return queryModifier.apply(session, selectQuerySql); // TODO is this required?
+    }
+
+    private PreparedStatement buildUnloadSql(ConnectorSession session, Connection connection, List<JdbcColumnHandle> columns, String redshiftSelectSql)
+            throws SQLException
+    {
+        String unloadSql = "UNLOAD ('%s') TO '%s' %s FORMAT PARQUET".formatted(
```

Comment:
We should add MAXFILESIZE 64MB to ensure that there is sufficient parallelism from multiple files

Reply:
I was about to ask you the ideal file size to set. Changed it to 64MB.